src/org/omegat/filters2/master/PluginUtils.java

```diff
@@ -72,8 +72,29 @@
 
     public static final String PLUGINS_LIST_FILE = "Plugins.properties";
 
-    enum PluginType {
-        FILTER, TOKENIZER, MARKER, MACHINETRANSLATOR, BASE, GLOSSARY, UNKNOWN
+    /**
+     * Plugin type definitions.
+     */
+    public enum PluginType {
+        FILTER("filter"),
+        TOKENIZER("tokenizer"),
+        MARKER("marker"),
+        MACHINETRANSLATOR("machinetranslator"),
+        BASE("base"),
+        GLOSSARY("glossary"),
+        DICTIONARY("dictionary"),
+        MISCELLANEOUS("miscellaneous"),
+        UNKNOWN("Undefined");
+
+        private final String typeValue;
```

Comment:
What is the need for `typeValue`? It looks like it's supposed to be a UI string, but that's not ideal because we can't translate them.

Instead of having UI strings as members of the enum class, the UI string should be looked up like `OStrings.getString("PLUGIN_TYPE_" + PluginType.FOO)`.

Reply:
I use `typeValue` to compare with `Plugin-Category` value in MANIFEST, and get corresponding enum object.

Reply:
I've adjusted relative functions to synthesize the policy.

> Instead of having UI strings as members of the enum class, the UI string should be looked up like OStrings.getString("PLUGIN_TYPE_" + PluginType.FOO).

I'd like to do this for future UI change.